apiserver/facades/client/application/application.go

```diff
@@ -2146,6 +2146,12 @@ func (api *APIBase) setApplicationConfig(arg params.ApplicationConfigSet) error
 		if err != nil {
 			return errors.Trace(err)
 		}
+
+		// We need a guard on the API server-side for direct API callers such as
+		// python-libjuju. Always default to the master branch.
```

Comment:
it is also older Juju clients (2.5) not just libjuju

Reply:
Added that detail.